controllers/ocirepository_controller.go

```diff
@@ -377,8 +389,8 @@ func (r *OCIRepositoryReconciler) getArtifactURL(ctx context.Context, obj *sourc
 
 // getTagBySemver call the remote container registry, fetches all the tags from the repository,
 // and returns the latest tag according to the semver expression.
-func (r *OCIRepositoryReconciler) getTagBySemver(ctx context.Context, url, exp string) (string, error) {
-	tags, err := crane.ListTags(url, r.craneOptions(ctx)...)
+func (r *OCIRepositoryReconciler) getTagBySemver(ctx context.Context, url, exp string, keychain authn.Keychain) (string, error) {
```

Comment:
When pulling a Helm Chart from an OCI registry, we also return the latest tag according to a semver expression.

The code is here: https://github.com/fluxcd/source-controller/blob/3dcb1420760d8aa0dfb5d496910fd3f9576aeb8a/internal/helm/repository/oci_chart_repository.go#L208

We may want to have the same underlying function for both implementation to ensure the logic remains consistent.

Adding the comment here to keep a trace, but does not have to be addressed in this PR.